elastic/kibana
# [Response Ops] Using data view service to fetch data views

Resolves https://github.com/elastic/kibana/issues/129914

## Summary

Using data view service to fetch index patterns instead of using the saved objects client directly. Previously, when using the saved objects client, we retrieved 1000 data views at a time and paginated if the total number of matching data views exceeded 1000. The data views service does not provide the ability to specify page number so functionality for this changes a bit as `loadIndexPatterns` will now only return up to 1000 matching data views. Discussed with the team and we felt that this should be sufficient, especially since the data views are filtered based on user input.

## To Verify

Verify that selecting index pattern for Index Threshold and ES Query rule type still work as expected. Also verify that selecting index pattern for ES Index connector type works as expected.

### Checklist
